### ISSUE TITLE
Location opening hours and Tariff start/end time: in local time or UTC?

### ISSUE THREAD

**Issue by RobertdeLeeuw-IHomer** [ocpi/ocpi#231]
OCPI 2.1.1 and 2.2 contains a couple of time fields that are String fields with a fixed format.

Location:
- RegularHours.period_begin
- RegularHours.period_end

Tariff:
- TariffRestrictions.start_time
- TariffRestrictions.end_time
also relevant: (when does a day start, depends on timezone)
- TariffRestrictions.start_date
- TariffRestrictions.end_date

Are these fields in local time or UTC?
Currently it is not defined.
DateTime is defined to be in UTC.
But as OpeningHours is typical for the EV driver, so human readable, it might make more sense to define them in local time.
But for TariffRestriction, it is more for systems to read. So there UTC might be more suitable.

Something to keep in mind:
The Location ExceptionalPeriod contains 2 fields that are DateTime format
- ExceptionalPeriod.period_begin
- ExceptionalPeriod.period_end
These are DateTime because the exceptions are expected to be exceptional days, not day of week, but say for example Christmas.

**Comment by Namoshek** [comment]
**TL;DR** My preferred solution is to support DateTime offsets in the form of `[+/-][hh]:[mm]` as in 
`2019-08-05T17:38:44+02:00`.

---

OCPI2.2-RC2 contains the following description of the `DateTime` type in section 16.2:
> All timestamps are formatted as string(25) following RFC 3339, with some additional limitations.
>
> All timestamps SHALL be in UTC. The absence of the timezone designator implies a UTC timestamp. Fractional seconds MAY be used.
>
> Example of how timestamps shall be formatted in OCPI, other formats/patterns are not allowed:
> 2015-06-29T20:39:09Z
> 2015-06-29T20:39:09
> 2016-12-29T17:45:09.2Z
> 2016-12-29T17:45:09.2
> 2018-01-01T01:08:01.123Z
> 2018-01-01T01:08:01.123
>
> **NOTE** +00:00 is not the same as UTC.

This means the following to me:
- `string(25)` can, as described in RFC 3339, fully contain a DateTime with offset like 
`2019-08-05T17:29:24+02:00` (25 characters) and is, implied by the NOTE, supported.
- If not stated otherwise through an offset, DateTimes are given and to be understood in UTC.

What I don't get is the NOTE, as +00:00 is actually the same as UTC. At least according to RFC 3339. Maybe this is a mistake?
> **4.3. Unknown Local Offset Convention**
> If the time in UTC is known, but the offset to local time is unknown, this can be represented with an offset of "-00:00".  This differs semantically from an offset of "Z" or "+00:00", which imply that UTC is the preferred reference point for the specified time. [...]

**Comment by webprofusion-chrisc** [comment]
From my experience all dates should be stored in UTC and only converted to local time when presented to the user (context specific). This is because of seasonal changes to daylight time saving (which would temporarily make the stored date/times wrong) and because in some countries users will span multiple time zones (sometime a difference of 30 minutes)  so the potential for confusion increases.

To make things more difficult your interest in date varies depending on why you are looking at the data - e.g. identifying which stations are currently closed (across a country which spans multiple time zones) .

**Comment by RobertdeLeeuw-IHomer** [comment]
Fixed: https://github.com/ocpi/ocpi/commit/cf2670443ba4d0cb6b2c4cebf578a6a2f1f53f48
and: https://github.com/ocpi/ocpi/commit/d1d424677bf399a781ed0d9b975b5c79cc8a2347
